src/Microsoft.DotNet.Wpf/src/WpfGfx/core/common/d3dutils.cpp

```diff
@@ -232,10 +232,21 @@ GetMinimalTextureDesc(
         }
 
         // If format is ok, restore HR from width/height check
-        if (hr == S_OK)
+        if (RenderOptions::IsGraphicHWAccelerationForRdpEnabled())
         {
-            hr = hrWH;
+            if(SUCCEEDED(hr))
```

Comment:
It would be great to be explicit here. You could also simplify the logic:

```C++
if (hr == S_OK || (hr == D3DOK_NOAUTOGEN && RenderOptions::IsGraphicHWAccelerationForRdpEnabled()))
{
    hr = hrWH;
}
```

I'd recommend doing the same sort of thing in the other blocks.